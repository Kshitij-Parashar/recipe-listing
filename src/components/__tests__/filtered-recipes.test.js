import React from 'react';
import { Provider } from 'react-redux';
import { Router } from 'react-router-dom';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import FilteredRecipes from '../filtered-recipes/filtered-recipes';
import '../../setupTests';

it('should render correctly <FilteredRecipes>', () => {
  const historyMock = {
    goBack: jest.fn(),
    listen: jest.fn(),
    location: { pathname: '' },
  };
  const mockStore = configureStore();

  const store = mockStore({
    recipes: {
      meals: [
        {
          idMeal: '52874',
          strMeal: 'Beef and Mustard Pie',
          strMealThumb:
            'https://www.themealdb.com/images/media/meals/sytuqu1511553755.jpg',
        },
        {
          idMeal: '52878',
          strMeal: 'Beef and Oyster pie',
          strMealThumb:
            'https://www.themealdb.com/images/media/meals/wrssvt1511556563.jpg',
        },
      ],
      loading: false,
      error: false,
    },
  });
  const wrapper = mount(
    <Provider store={store}>
      <Router history={historyMock}>
        <FilteredRecipes />
      </Router>
    </Provider>
  );
  expect(wrapper).toMatchSnapshot();
});
