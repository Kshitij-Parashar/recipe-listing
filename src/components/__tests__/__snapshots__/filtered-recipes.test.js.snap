// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly <FilteredRecipes> 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Router
    history={
      Object {
        "goBack": [MockFunction],
        "listen": [MockFunction] {
          "calls": Array [
            Array [
              [Function],
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        },
        "location": Object {
          "pathname": "",
        },
      }
    }
  >
    <Connect(RecipeByCategory)>
      <RecipeByCategory
        error={false}
        loadRecipes={[Function]}
        loading={false}
        setCategory={[Function]}
      >
        <h1
          className="heading center"
        >
           Recipes
        </h1>
        <div
          className="recipes-by-category"
        >
          <button
            className="back-button"
            onClick={[Function]}
          >
            &lt;&lt; 
            Back
          </button>
          <RecipeList
            error={false}
            loading={false}
            recipes={Array []}
          >
            <h2>
              No recipes found.
            </h2>
          </RecipeList>
        </div>
      </RecipeByCategory>
    </Connect(RecipeByCategory)>
  </Router>
</Provider>
`;
